web3.md

```diff
@@ -1058,5 +1060,45 @@ NOGAS Mode
     rule <k> #validateTx TXID => loadCallState TXID ~> #executeTx TXID ~> #makeTxReceipt TXID ... </k>
          <mode> NOGAS </mode>
      [priority(25)]
+```
+
+Collecting Coverage Data
+------------------------
+
+- `<execPhase>` cell is used to differentiate between the generated code used for contract deployment and the bytecode of the contract.
+- `<opcodeCoverage>` cell is a map which stores the program counters which were hit during the execution of a program. The key, named `CoverageIdentifier`, contains the hash of the bytecode which is executed, and the phase of the execution.
+
+```k
+    syntax Phase ::= ".Phase"
+                   | "CONSTRUCTOR"
+                   | "RUNTIME"
+
+    syntax CoverageIdentifier ::= "{" Int "|" Phase "}"
+
+    rule <k> #mkCall _ _ _ _ _ _ _ ... </k>
+         <execPhase> ( EPHASE => RUNTIME ) </execPhase>
+      requires EPHASE =/=K RUNTIME
+      [priority(25)]
+
+    rule <k> #mkCreate _ _ _ _ ... </k>
+         <execPhase> ( EPHASE => CONSTRUCTOR ) </execPhase>
+      requires EPHASE =/=K CONSTRUCTOR
+      [priority(25)]
+
+    rule <k> #initVM ... </k>
+         <opcodeCoverage> OC => OC [ {keccak(CODE) | EPHASE} <- .Set ] </opcodeCoverage>
+         <execPhase> EPHASE </execPhase>
+         <program> CODE </program>
+      requires notBool {keccak(CODE) | EPHASE} in_keys(OC)
+      [priority(25)]
```

Comment:
If we do a call `A -> B -> A`, this will lose all coverage information from the first call of `A` when doing the second call to `A`.

Reply:
Nvm, we have the condition `notBool ... in_keys`